tests/Aspire.Hosting.RabbitMQ.Tests/RabbitMQFunctionalTests.cs

```diff
@@ -0,0 +1,195 @@
+                        channel.BasicPublish(
+                            exchange: string.Empty,
+                            queueName,
+                            props,
+                            body);
+                    }
+                }
+                finally
+                {
+                    // Stops the container, or the Volume/mount would still be in use
+                    await app.StopAsync();
+                }
+            }
+
+            var builder2 = CreateDistributedApplicationBuilder();
+            var passwordParameter2 = builder2.AddParameter("pwd");
+            builder2.Configuration["Parameters:pwd"] = password;
+
+            var rabbitMQ2 = builder2.AddRabbitMQ("rabbitMQ", password: passwordParameter2);
+
+            if (useVolume)
+            {
+                rabbitMQ2.WithDataVolume(volumeName);
+            }
+            else
+            {
+                rabbitMQ2.WithDataBindMount(bindMountPath!);
+            }
+
+            using (var app = builder2.Build())
+            {
+                await app.StartAsync();
+                try
+                {
+                    var hb = Host.CreateApplicationBuilder();
+                    hb.Configuration[$"ConnectionStrings:{rabbitMQ2.Resource.Name}"] = await rabbitMQ2.Resource.ConnectionStringExpression.GetValueAsync(default);
+                    hb.AddRabbitMQClient(rabbitMQ2.Resource.Name);
+
+                    using (var host = hb.Build())
+                    {
+                        await host.StartAsync();
+
+                        var connection = host.Services.GetRequiredService<IConnection>();
+
+                        using var channel = connection.CreateModel();
+                        const string queueName = "hello";
+                        channel.QueueDeclare(queueName, durable: true, exclusive: false);
+
+                        var result = channel.BasicGet(queueName, true);
+                        Assert.Equal("Hello World!", Encoding.UTF8.GetString(result.Body.Span));
+                    }
+                }
+                finally
+                {
+                    // Stops the container, or the Volume/mount would still be in use
+                    await app.StopAsync();
+                }
+            }
+        }
+        finally
+        {
+            if (volumeName is not null)
+            {
+                DockerUtils.AttemptDeleteDockerVolume(volumeName);
+            }
+
+            if (bindMountPath is not null)
+            {
+                try
+                {
+                    Directory.Delete(bindMountPath, recursive: true);
+                }
+                catch
+                {
+                    // Don't fail test if we can't clean the temporary folder
+                }
+            }
+        }
+    }
+
+    private TestDistributedApplicationBuilder CreateDistributedApplicationBuilder()
+    {
+        var builder = TestDistributedApplicationBuilder.Create();
```

Comment:
```suggestion
        var builder = TestDistributedApplicationBuilder.CreateWithTestContainerRegistry();
```

Reply:
Good call. This was left over from trying to debug something.